pkg/network/setup/netstat.go

```diff
@@ -67,106 +66,78 @@ func (c *NetStat) CachePodInterfaceVolatileData(vmi *v1.VirtualMachineInstance,
 	c.podInterfaceVolatileCache.Store(vmiInterfaceKey(vmi.UID, ifaceName), data)
 }
 
+// UpdateStatus calculates the vmi.Status.Interfaces based on the following data sets:
+// - Pod interface cache: interfaces data (IP/s) collected from the cache (which was populated during the network setup).
+// - domain.Spec: interfaces configuration as seen by the (libvirt) domain.
+// - domain.Status.Interfaces: interfaces reported by the guest agent (empty if Qemu agent not running).
 func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domain) error {
 	if domain == nil {
 		return nil
 	}
 
-	interfacesStatus, err := c.ifacesStatusFromPod(vmi)
+	interfacesStatus, err := c.ifacesStatusFromPodCache(vmi)
 	if err != nil {
 		return err
 	}
 	vmi.Status.Interfaces = interfacesStatus
 
-	if len(domain.Spec.Devices.Interfaces) > 0 || len(domain.Status.Interfaces) > 0 {
-		// This calculates the vmi.Status.Interfaces based on the following data sets:
-		// - vmi.Status.Interfaces - interfaces data (IP/s) collected from the pod (non-volatile cache).
-		// - domain.Spec - interfaces form the Spec
-		// - domain.Status.Interfaces - interfaces reported by guest agent (empty if Qemu agent not running)
-		newInterfaces := []v1.VirtualMachineInstanceNetworkInterface{}
-
-		existingInterfaceStatusByName := map[string]v1.VirtualMachineInstanceNetworkInterface{}
-		for _, existingInterfaceStatus := range vmi.Status.Interfaces {
-			if existingInterfaceStatus.Name != "" {
-				existingInterfaceStatusByName[existingInterfaceStatus.Name] = existingInterfaceStatus
+	if len(domain.Spec.Devices.Interfaces) > 0 {
+		vmiInterfacesStatusByName := map[string]*v1.VirtualMachineInstanceNetworkInterface{}
+		for index, ifaceStatus := range vmi.Status.Interfaces {
+			vmiInterfacesStatusByName[ifaceStatus.Name] = &vmi.Status.Interfaces[index]
+		}
+
+		for _, domainSpecInterface := range domain.Spec.Devices.Interfaces {
+			interfaceMAC := domainSpecInterface.MAC.MAC
+			domainIfaceAlias := domainSpecInterface.Alias.GetName()
+
+			if interfaceStatus, exists := vmiInterfacesStatusByName[domainIfaceAlias]; exists {
+				interfaceStatus.MAC = interfaceMAC
+				interfaceStatus.InfoSource = netvmispec.InfoSourceDomain
+			} else {
+				vmi.Status.Interfaces = append(vmi.Status.Interfaces, v1.VirtualMachineInstanceNetworkInterface{
+					Name:       domainIfaceAlias,
+					MAC:        interfaceMAC,
+					InfoSource: netvmispec.InfoSourceDomain,
+				})
 			}
 		}
+	}
 
-		domainInterfaceStatusByMac := map[string]api.InterfaceStatus{}
+	if len(domain.Status.Interfaces) > 0 {
```

Comment:
Yes, as described in the method description: `domain.Status.Interfaces: interfaces reported by the guest agent (empty if Qemu agent not running).`

Reply:
or was presented but later disabled (since virt-launcher has caching of the guest-agent report)

we might want to consider changing this behavior in the future.
If guest agent is stopped, the cache will still report for ever the last known state